### DISCUSSION TITLE
Allow Mean Planner to create Recipe Timeline events

### DISCUSSION THREAD

**Discussion by shiruken** [mealie-recipes/mealie#2037] [Feature Request]
**Before submitting this feature request I have**
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
The "I Made This" (#1801) and Recipe Timeline (#1831) features should optionally allow integration with the Meal Planner. As currently implemented, a user has to specify dates twice, once in the Meal Planner and once using the recipe's "I Made This" button. There should be an option in the Meal Planner to allow for the automatic creation of Recipe Timeline events.

**Comment by michael-genson** [comment]
A few questions about this, since I'm not a regular meal planner user (also open to anyone else who sees this):
1. Would you expect the event to be created immediately? (i.e. there is an event set in the future) or would you expect it to be created automatically on your behalf later? (i.e. on the day of your meal plan)
2. Would you expect the subject of the event to be different, or the same as when you click "I Made This"? For instance, would you want it to be something like "Mealie User made this for Breakfast" or similar.
3. When you choose "I Made This" you have the option to include additional text (e.g. how it turned out). Would you want this option when creating a meal plan? If yes, what would you use it for? (asking so we can possibly add a hint, similar to the one on the I Made This dialog).
4. Is there any scenario you _wouldn't_ want to add this to the recipe timeline? (asking because we could just make it automatic so we don't even prompt you with the option)

**Reply by shiruken** [reply]
1. Creating the event on the day of the meal plan seems preferable to me, although I could see scenarios where a user might want to know that a recipe is already planned for a future date.
2. Appending the entry type (e.g. Breakfast, Lunch, Dinner, Side) also makes sense although could be excluded for simplicity.
3. The additional text seems unnecessary in this context since details about how it turned out are unknown until the recipe is actually made. If the user wants to add these details, they should probably just edit the the timeline event itself afterwards.
4. I think we would want to exclude "Note Only" mean plan entries (i.e. there has to be an associated recipe) but otherwise would want automatic timeline entries. Perhaps a setting could be added to the Meal Plan Settings page (`/group/mealplan/settings`) to toggle automatic Recipe Timeline event creation.

**Comment by michael-genson** [comment]
See https://github.com/hay-kot/mealie/pull/2050
